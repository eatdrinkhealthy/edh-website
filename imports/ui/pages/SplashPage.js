import React from "react";
import { defaultPageStyle } from "../styles";
import CompanyName from "../components/CompanyName";
import Pitch from "../components/Pitch";
import ComingSoon from "../components/ComingSoon";
import Logo from "../components/Logo";

const SplashPage = () => (
  <div className={defaultPageStyle.className}>
    <CompanyName />
    <Pitch />
    <ComingSoon />
    <Logo />
  </div>
);

SplashPage.style = defaultPageStyle;

export default SplashPage;
