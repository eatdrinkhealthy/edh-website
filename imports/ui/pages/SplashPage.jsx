import React from "react";
import Header from "../components/Header";
import Footer from "../components/Footer";
import Preview from "../components/Preview";

const SplashPage = () => (
  <div>
    <Header />
    <Preview />
    <Footer />
  </div>
);

export default SplashPage;
