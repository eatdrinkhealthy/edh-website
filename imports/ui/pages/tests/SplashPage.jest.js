/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */

import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import SplashPage from "../SplashPage";

describe("<SplashPage />", function () {
  it("render matches snapshot", function () {
    const tree = renderer.create(<SplashPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("css matches snapshot", function () {
    expect(shallow(<SplashPage />).hasClass(SplashPage.style.className)).toBe(true);
    expect(JSON.stringify(SplashPage.style, null, 2)).toMatchSnapshot();
  });

  it("contains sections company name, pitch, coming soon, logo, preview ", function () {
    const wrapper = shallow(<SplashPage />);
    expect(wrapper.find("CompanyName").length).toEqual(1);
    expect(wrapper.find("Pitch").length).toEqual(1);
    expect(wrapper.find("ComingSoon").length).toEqual(1);
  });
});
