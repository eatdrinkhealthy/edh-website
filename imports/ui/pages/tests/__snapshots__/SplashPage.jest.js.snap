exports[`<SplashPage /> css matches snapshot 1`] = `
"{
  \"rules\": {
    \"background\": [
      \"#048ec5\",
      \"-moz-linear-gradient(-45deg, #048ec5 0%, #2ecc71 100%) fixed\",
      \"-webkit-linear-gradient(-45deg, #048ec5 0%,#2ecc71 100%) fixed\",
      \"-webkit-linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\",
      \"-moz-linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\",
      \"linear-gradient(135deg, #048ec5 0%,#2ecc71 100%) fixed\"
    ],
    \"filter\": \"progid:DXImageTransform.Microsoft.gradient( startColorstr=\'#048ec5\', endColorstr=\'#2ecc71\',GradientType=1 )\",
    \"color\": \"white\",
    \"padding\": \"30px\"
  },
  \"className\": \"css-wlbfil\"
}"
`;

exports[`<SplashPage /> render matches snapshot 1`] = `
<div
  className="css-wlbfil">
  <div
    className="css-nkpr7m" />
  <div
    className="css-1dl4n80">
    A platform to find, share, and discuss healthy places to shop and eat.
  </div>
  <div
    className="css-cdesk3">
    Coming Soon
  </div>
  <div
    className="css-o94tcm" />
</div>
`;
